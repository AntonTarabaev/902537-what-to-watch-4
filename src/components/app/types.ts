import {Comment, Film} from "@root/types";
import {AppActionTypes} from "@constants/action-types";

export interface DataState {
  films: Film[];
  promo: Film;
  comments: Comment[];
  isLoaded: boolean;
}

interface SetFilmsAction {
  type: typeof AppActionTypes.SET_FILMS;
  payload: Film[];
}

interface SetPromoAction {
  type: typeof AppActionTypes.SET_PROMO;
  payload: Film;
}

interface SetCommentsAction {
  type: typeof AppActionTypes.SET_COMMENTS;
  payload: Comment[];
}

interface SetLoadedStatusAction {
  type: typeof AppActionTypes.SET_LOADED_STATUS;
  payload: boolean;
}

interface SetFilmIsFavoriteAction {
  type: typeof AppActionTypes.SET_FILM_IS_FAVORITE;
  payload: Film;
}

export type DataActions = SetCommentsAction | SetFilmIsFavoriteAction | SetFilmsAction | SetPromoAction | SetLoadedStatusAction;
