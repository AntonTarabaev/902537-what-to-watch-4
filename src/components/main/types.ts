import {GenresListActionTypes, MainPageActionTypes} from "@constants/action-types";

export interface MainPageState {
  filterGenre: string;
  showingFilmsCount: number;
}

interface IncrementShowingFilmsAction {
  type: typeof MainPageActionTypes.INCREMENT_SHOWING_FILMS;
  payload: number;
}

interface ChangeActiveGenreAction {
  type: typeof GenresListActionTypes.CHANGE_GENRE_FILTER;
  payload: string;
}

interface ResetShowingFilmsAction {
  type: typeof GenresListActionTypes.RESET_SHOWING_FILMS;
  payload: number;
}

export type MainPageActions = IncrementShowingFilmsAction | ChangeActiveGenreAction | ResetShowingFilmsAction;
