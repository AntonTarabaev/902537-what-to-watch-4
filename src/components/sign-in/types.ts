import {AuthorizationStatus, Film, User} from "@root/types";
import {UserActionTypes} from "@constants/action-types";

export interface UserState {
  authorizationStatus: AuthorizationStatus;
  userData: User;
  favoriteFilms: Film[];
}

interface RequireAuthorizationAction {
  type: typeof UserActionTypes.REQUIRED_AUTHORIZATION;
  payload: AuthorizationStatus;
}

interface SetUserDataAction {
  type: typeof UserActionTypes.SET_USER_DATA;
  payload: User;
}

interface SetFavoriteAction {
  type: typeof UserActionTypes.SET_FAVORITE;
  payload: Film[];
}

export type UserActions = RequireAuthorizationAction | SetUserDataAction | SetFavoriteAction;
