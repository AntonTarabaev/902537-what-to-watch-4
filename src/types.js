export const Film = PropTypes.shape({
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  bgImage: PropTypes.string.isRequired,
  previewImage: PropTypes.string.isRequired,
  releaseYear: PropTypes.number.isRequired,
  genre: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  ratingVotes: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  actors: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  preview: PropTypes.string.isRequired,
  duration: PropTypes.number.isRequired,
  video: PropTypes.string.isRequired,
  isFavorite: PropTypes.bool.isRequired,
}).isRequired;

export const Comment = PropTypes.shape({
  author: PropTypes.string.isRequired,
  comment: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired,
  rating: PropTypes.number.isRequired,
}).isRequired;
