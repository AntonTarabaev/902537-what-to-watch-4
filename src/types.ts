export enum AuthorizationStatus {
  AUTH = `AUTH`,
  NO_AUTH = `NO_AUTH`,
}

export interface Film {
  id: string;
  title: string;
  poster: string;
  bgImage: string;
  previewImage: string;
  releaseYear: number;
  genre: string;
  rating: number;
  ratingVotes: number;
  director: string;
  description: string;
  actors: string[];
  preview: string;
  duration: number;
  video: string;
  isFavorite: boolean;
}

export interface Comment {
  id: number;
  author: string;
  authorId: number;
  comment: string;
  date: Date;
  rating: number;
}
