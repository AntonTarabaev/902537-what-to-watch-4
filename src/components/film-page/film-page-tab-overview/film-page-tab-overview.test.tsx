import FilmPageTabOverview from "@components/film-page/film-page-tab-overview/film-page-tab-overview";
import {Film} from "@root/types";
import * as renderer from "react-test-renderer";

const film: Film = {
  id: `573489`,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewImage: `the-grand-budapest-hotel-poster.jpg`,
  bgImage: `path`,
  releaseYear: 1999,
  genre: `Drama`,
  rating: 8.2,
  ratingVotes: 197,
  director: `Some One`,
  duration: 123,
  description: `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`,
  actors: [
    `Robert Rodrigues`,
    `Takeshi Kitano`,
    `Michael Caine`,
    `Robert De Niro`,
    `Tom Hanks`,
  ],
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  video: `path`,
  isFavorite: true,
};

it(`Should FilmPageTabOverview render correctly`, () => {
  const tree = renderer.create(
      <FilmPageTabOverview
        film={film}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
