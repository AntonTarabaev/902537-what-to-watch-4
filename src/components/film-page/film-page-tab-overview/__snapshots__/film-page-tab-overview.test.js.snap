// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmPageTabDetails render correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.2
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        197
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
    </p>
    <p>
      Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Some One
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Robert Rodrigues, Takeshi Kitano, Michael Caine, Robert De Niro and others
      </strong>
    </p>
  </div>,
]
`;
